healthyregions/SDOHPlace
# Add matched theme icon and enable all subject filters

This PR addresses #287, #296, and #297. It adds matching MUI icons to subjects in the result list and detail panel and enables the filter operation when someone clicks on them.

Additionally, this PR includes a fix for #305. When a user opens the detailed view and then applies a filter that triggers the "no results" view, the map view will be restored instead of leaving the page blank.

## How to Test
1. Navigate to the search page. You should see that all result item header images have been updated to match their corresponding subject:
<img width="1753" alt="Screenshot 2024-09-20 at 5 03 37 PM" src="https://github.com/user-attachments/assets/06933d7c-9079-415a-a9a1-4443193718a4">


2. Click on an item with the "Compose" icon. You should see more than one subject tag for this item in the detailed view, with the icon in the title also updated:
<img width="1411" alt="Screenshot 2024-09-20 at 5 07 11 PM" src="https://github.com/user-attachments/assets/e5ec3166-ba97-4aa4-ad9e-6d0f7ee42ca6">

3. Click on an item with another subject icon. You should see the corresponding subject tag for this item in the detailed view:
<img width="1565" alt="Screenshot 2024-09-20 at 5 08 12 PM" src="https://github.com/user-attachments/assets/b4cc4f7b-0e36-43b2-8106-0d3b874224e0">


4. Click the subject icon in the detailed view. The result list should now be filtered by this subject:
<img width="1393" alt="Screenshot 2024-09-20 at 5 08 34 PM" src="https://github.com/user-attachments/assets/62ebfa1d-6d32-43ea-8586-ce6ba31f432c">


5. While in the detailed view, apply a spatial resolution filter that does not match this result item. The detailed view should be replaced by the Map view, as this item will no longer be in the result list:
<img width="1793" alt="Screenshot 2024-09-20 at 5 12 42 PM" src="https://github.com/user-attachments/assets/727198b2-80ff-48bd-80cc-a85ec14165d5">

6. Search for "a" in the search box. You should see the "no results" view with all matched subject buttons displayed. Note that I’ve simply placed all subject buttons here for now; we can adjust this after improving the recommendation system:
<img width="2013" alt="Screenshot 2024-09-20 at 5 13 05 PM" src="https://github.com/user-attachments/assets/8a5ba88a-f866-44bf-beb7-ee6dee0b7b72">


7. Click "Compose," and you should see the same result as in step 2. During these operations, the user can always click "reset all filters" to return to the original state without interfering with the query results:
<img width="906" alt="Screenshot 2024-09-20 at 5 15 39 PM" src="https://github.com/user-attachments/assets/e5540ae4-90cb-427b-b34a-d60ca67f4585">
